mantis-publish/mantis-publish-core/src/main/java/io/mantisrx/publish/AbstractSubscriptionTracker.java

```diff
@@ -66,52 +72,51 @@ public AbstractSubscriptionTracker(MrePublishConfiguration mrePublishConfigurati
         this.staleSubscriptionRemovedCount = SpectatorUtils.buildAndRegisterCounter(registry, "staleSubscriptionRemovedCount");
     }
 
-    void propagateSubscriptionChanges(Set<MantisServerSubscription> prev, Set<MantisServerSubscription> curr) {
-        Set<MantisServerSubscription> prevSubsNotInCurr = new HashSet<>(prev);
-        prevSubsNotInCurr.removeAll(curr);
-        prevSubsNotInCurr.stream().forEach(subToRemove -> {
-            try {
-                Optional<Subscription> subscription = SubscriptionFactory.getSubscription(subToRemove.getSubscriptionId(), subToRemove.getQuery());
-                if (subscription.isPresent()) {
-                    streamManager.removeStreamSubscription(subscription.get());
-                } else {
-                    LOG.warn("unexpected to find invalid subscription to remove {}", subToRemove);
-                }
-            } catch (Throwable t) {
-                if (LOG.isDebugEnabled()) {
-                    LOG.debug("failed to remove subscription {}", subToRemove, t);
+    void propagateSubscriptionChanges(String streamName, Set<MantisServerSubscription> curr) {
+        Set<String> previousIds = getCurrentSubIds(streamName);
+
+        for (MantisServerSubscription newSub : curr) {
+            // Add new subscription not present previously
+            if (!previousIds.contains(newSub.getSubscriptionId())) {
+                try {
+                    Optional<Subscription> subscription = SubscriptionFactory.getSubscription(newSub.getSubscriptionId(), newSub.getQuery());
+                    if (subscription.isPresent()) {
+                        streamManager.addStreamSubscription(subscription.get());
+                    } else {
+                        LOG.info("will not add invalid subscription {}", newSub);
```

Comment:
I'm assuming this should never happen? Do you want to may be increase the log level on this?